docs/jwt-auth.md

```diff
@@ -0,0 +1,153 @@
+  "nbf": 1660119383,
+  "sub": "0x556f8BE42f76c01F960f32CB1936D2e0e0Eb3F4D"
+}
+```
+
+Every field above except for `bot-id` is defined in the [JWT standard](https://auth0.com/docs/secure/tokens/json-web-tokens/json-web-token-claims#registered-claims). Additional claims can also be encoded in the JWT token using the first input `claims`, you also specify when the token should expire as the second input:
+
+``` typescript
+const initialize: Initialize = async () => {
+  const token = await fetchJwtToken({key: "value"}, Date.now())
+  const decodedTokenData = decodeJwtToken(token)
+
+  ...
+}
+```
+
+``` python
+def initialize():
+    token = fetch_jwt_token({'key': 'value'})
+    decoded_token_data = decode_jwt_token(token)
+
+    ...
+```
+
+which will result in a `decodedTokenData` like below:
+
+```
+{
+  "bot-id": "0x13k387b37769ce24236c403e76fc30f01fa774176e1416c861yfe6c07dfef71f",
+  "exp": 1660119443,
+  "iat": 1660119413,
+  "jti": "qkd5cfad-1884-11ed-a5c9-02420a639308",
+  "nbf": 1660119383,
+  "sub": "0x556f8BE42f76c01F960f32CB1936D2e0e0Eb3F4D",
+  "key": "value"
+}
+```
+
+A useful tool for manually decoding tokens is [this web app](https://jwt.io/) provided by auth0.
+
+## Detection Bot Authentication Example
+
+Imagine you have a backend service that exposes an API endpoint that you would like your detection bot be authorized to use. Here is an example express api endpoint that uses a JWT for authentication at the host name of `external-api`. This server has a GET endpoint the returns data from an applications database (this example does not add any additional claims):
+
+``` typescript
+import express, { Request, Response } from "express";
+import { decodeJwtToken } from "forta-agent";
+
+const PATH = `/example-endpoint`;
+const PORT = 3000;
+
+const app = express();
+const requestRouter = express.Router();
+
+interface JwtPayload {
+  botId: string,
+  exp: number,
+  iat: number,
+  jti: string,
+  nbf: number,
+  sub: string
+}
+
+// Define /example-endpoint
+requestRouter.get(PATH, async (request: Request, response: Response) => {
+
+    // Assuming the JWT token is passed in the header "x-access-token". You can choose a different method to pass the JWT
+    const token = req.headers["x-access-token"];
+
+    const data: JwtPayload = decodeJwtToken(token);
+
+    // If you add additional claims such as api keys or secrets your verification logic can use those as well
+    if(data && data.botId === "YOUR_BOT_ID") {
+      // Fetch data from database and return it in response
+    } else {
+      // return 401 error for invalid JWT
+    }
+})
+
+app.use("/", requestRouter)
+
+app.listen(PORT, () => {
+  console.log(`Server starting`)
+})
+
+```
+
+Your bot could call this endpoint like the following (using axios for external calls):
+
+``` typescript
+
+let storedToken: string
+
+const initialize: Initialize = async () => {
+  const token = await fetchJwtToken({key: "value"}, Date.now())
```

Comment:
we can just assign to `storedToken` on this line. also, the expiry should probably be some point in the future

Reply:
Updated